src/snowflake/cli/_plugins/stage/manager.py

```diff
@@ -480,10 +482,6 @@ def _get_files_list_from_stage(
         self, stage_path: StagePath, pattern: str | None = None
     ) -> List[str]:
         files_list_result = self.list_files(stage_path, pattern=pattern).fetchall()
-
-        if not files_list_result:
-            raise ClickException(f"No files found on stage '{stage_path}'")
```

Comment:
This was causing the error. Exception moved to the function which calls this helper. Now the helper can return an empty list.